LiteCore/tests/QueryParserTest.cc

```diff
@@ -697,6 +697,20 @@ TEST_CASE_METHOD(QueryParserTest, "QueryParser Vector Search", "[Query][QueryPar
           == "SELECT key, sequence FROM kv_default AS _doc JOIN \"kv_default:vector:vecIndex\" AS vector1 ON "
              "vector1.rowid = _doc.rowid WHERE (vector1.vector LIKE encode_vector((array_of(12, 34))) AND _doc.key > "
              "'x') AND (_doc.flags & 1 = 0) ORDER BY vector1.distance");
+    // Pure vector search with default max_results:
+    CHECK(parse("['SELECT', {WHERE: ['VECTOR_MATCH()', 'vecIndex', ['[]', 12, 34]],"
+                "ORDER_BY: [ ['VECTOR_DISTANCE()', 'vecIndex'] ]}]")
+          == "SELECT key, sequence FROM kv_default AS _doc JOIN (SELECT rowid, distance FROM "
+             "\"kv_default:vector:vecIndex\" WHERE vector LIKE encode_vector(array_of(12, 34)) LIMIT 3) AS vector1 ON "
+             "vector1.rowid = _doc.rowid WHERE (true) AND (_doc.flags & 1 = 0) ORDER BY vector1.distance");
+    // Pure vector search, value of LIMIT is passed to VECTOR_MATCH as max_results.
+    CHECK(parse("['SELECT', {WHERE: ['VECTOR_MATCH()', 'vecIndex', ['[]', 12, 34]],"
```

Comment:
Isn't this identical to the check at line 678, only with limit 5 instead of 3?

Reply:
Right, but 3 is the magic default. That is, max_results will be 3 regardless of the LIMIT clause